nodes/number/curve_mapper.py

```diff
@@ -132,15 +132,25 @@ def sv_copy(self, node):
 
     def free(self):
         m = bpy.data.node_groups.get(node_group_name)
-        n_id = self.node_id
-        node = m.nodes['RGB Curves'+n_id]
+        node = m.nodes[self._get_curve_node_name()]
         m.nodes.remove(node)
 
+    def _get_curve_node_name(self):
+        n_id = self.node_id
+        return 'RGB Curves'+n_id
+
     def process(self):
         inputs = self.inputs
         outputs = self.outputs
-        n_id = self.node_id
-        evaluate = get_evaluator(node_group_name, 'RGB Curves'+n_id)
+        curve_node_name = self._get_curve_node_name()
+        evaluate = get_evaluator(node_group_name, curve_node_name)
+        curve_node = get_valid_node(node_group_name, curve_node_name, CURVE_NODE_TYPE)
+
+        curve = SvScalarFunctionCurve(evaluate)
+        curve.u_bounds = (curve_node.mapping.clip_min_x, curve_node.mapping.clip_max_x)
+        if 'Curve' in self.outputs:
+            self.outputs['Curve'].sv_set([curve])
```

Comment:
`if 'Curve' in self.outputs and self.outputs['Curve'].is_linked` could be before the curve is created

Reply:
Ok, I agree. These lines are really fast, but the code will be cleaner.